### PR TITLE
Copy/paste single elements

### PR DESCRIPTION
**Pull request by Broxzier**
This PR adds a copy/paste ability to the tile inspector, a feature requested on NEDesigns. One element can be copied to the in-game clipboard at a time, and the clipboard will get cleared when a new map is loaded.

### CODE REVIEW

File: src/windows/tile_inspector.c [hunk 1]
```diff
@@ -708,6 +720,26 @@ static void window_tile_inspector_sort_elements(rct_window *w) {
...
 }
 
+static void window_tile_inspector_copy_element(rct_window *w) {
+	// Copy value, in case the element gets moved
+	tileInspectorCopiedElement = *window_tile_inspector_get_selected_element(w);
+	windowTileInspectorElementCopied = true;
+}
+
+static void window_tile_inspector_paste_element(rct_window *w) {
+	rct_map_element *const pastedElement = map_element_insert(windowTileInspectorTileX, windowTileInspectorTileY, tileInspectorCopiedElement.base_height, 0);
```
Comment (Broxzier) [inline src/windows/tile_inspector.c]: This probably returns `NULL` when the limit has been reached. I've got to test this out, and display an error when that happens.

File: src/windows/tile_inspector.c [hunk 2]
```diff
@@ -570,6 +578,10 @@ void window_tile_inspector_open() {
 	window_tile_inspector_auto_set_buttons(window);
 }
 
+void window_tile_inspector_clear_clipboard() {
```
Comment (Gymnasiast) [inline src/windows/tile_inspector.c]: Opening braces of functions should go onto their own line. (I'm leaving this comment once, but it applies to every instance in this PR, of course.)

**Comment by duncanspumpkin** [comment]
Your copy and paste images, did you make them yourself?

**Comment by Broxzier** [comment]
@duncanspumpkin: Yes, I made the copy icon from scratch, and the paste one based on: http://icons.veryicon.com/ico/Application/Toolbar%20Icons/Paste.ico First I simply applied the game pallet to the ico file to get the correct colours, then copied one page from the copy icon onto it.

File: src/windows/tile_inspector.c (hunk 2 above)
Comment (Broxzier) [inline src/windows/tile_inspector.c]: @Gymnasiast Whoops, will fix it when I get home.

**Comment by Broxzier** [comment]
In case anyone is willing to test this, the 64-bit build can be [downloaded here](https://ci.appveyor.com/api/buildjobs/c3ydmx191vsxstny/artifacts/artifacts%2Fopenrct2-portable-release-x64-5a2073a.zip). As this was quite straightforward I don't expect many issues. If someone can take a look at where the clipboard gets cleared, then that would be nice. I tried to find all locations where a new park is loaded (and that way found lots of duplicated code).

**Comment by Margen67** [comment]
Pasted park entrances have an ID of -1 and crash the game.

**Comment by Broxzier** [comment]
All non-central park entrances have a park index of -1, and I am unable to reproduce this crash. Everything seems to work fine, even when I make a guest make contact with the entrance.

**Comment by Margen67** [comment]
They have to enter through it not just touch it

**Comment by Broxzier** [comment]
@Margen67: I'm still unable to reproduce the crash. Park entrance is indeed -1, because it's looking for the park entrance array using its position as the key, but it can't find anything, hence why it's -1.
Setup example
All guests in the picture walked through it. I'll keep on trying different things, but if you can provide an example, that would be helpful.
Even leaving through a copied entrance seems to work fine:
Entrance

I'm not entirely sure if this is really a bug to do with the tile inspector, it sounds more like something that has to do with the peeps (code I barely touched).

**Comment by Broxzier** [comment]
With the last fix, the game does not crash anymore (they turn around when reaching it), however guests are able to exit through it, and then will get stuck if there is no path leading from the copied entrance to their entry point. Personally I don't think this is important enough to be fixed (at this point at least).

**Pull request merged**